### ISSUE TITLE
Error installing with nightly 

### ISSUE THREAD

**Issue by dwillie** [Manishearth/rust-clippy#992]
Error installing using `rustup run nightly cargo install clippy` on debian jesse and OSX 10.11.3
```
vagrant@debian-jessie:~$ rustup run nightly cargo -V
cargo 0.12.0-nightly (510920b 2016-06-06)
vagrant@debian-jessie:~$ rustup run nightly rustc -V
rustc 1.11.0-nightly (763f9234b 2016-06-06)
```
```
vagrant@debian-jessie:~$ rustup run nightly cargo install clippy
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading clippy v0.0.74
 Downloading rustc-serialize v0.3.19
 Downloading quine-mc_cluskey v0.2.2
 Downloading clippy_lints v0.0.74
 Downloading semver v0.2.3
 Downloading unicode-normalization v0.1.2
 Downloading regex-syntax v0.3.3
 Downloading toml v0.1.30
 Downloading matches v0.1.2
 Downloading nom v1.2.3
   Compiling rustc-serialize v0.3.19
   Compiling unicode-normalization v0.1.2
   Compiling quine-mc_cluskey v0.2.2
   Compiling matches v0.1.2
   Compiling nom v1.2.3
   Compiling semver v0.2.3
   Compiling regex-syntax v0.3.3
   Compiling toml v0.1.30
   Compiling clippy_lints v0.0.74
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:16: 185:19 error: mismatched types [E0308]
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185         val == 0.0 || val == ::std::f64::INFINITY || val == ::std::f64::NEG_INFINITY
                                                                                                   ^~~
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:16: 185:19 help: run `rustc --explain E0308` to see a detailed explanation
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:16: 185:19 note: expected type `rustc_const_math::ConstFloat`
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:16: 185:19 note:    found type `_`
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:30: 185:50 error: mismatched types [E0308]
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185         val == 0.0 || val == ::std::f64::INFINITY || val == ::std::f64::NEG_INFINITY
                                                                                                                 ^~~~~~~~~~~~~~~~~~~~
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:30: 185:50 help: run `rustc --explain E0308` to see a detailed explanation
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:30: 185:50 note: expected type `rustc_const_math::ConstFloat`
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:30: 185:50 note:    found type `f64`
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:61: 185:85 error: mismatched types [E0308]
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185         val == 0.0 || val == ::std::f64::INFINITY || val == ::std::f64::NEG_INFINITY
                                                                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:61: 185:85 help: run `rustc --explain E0308` to see a detailed explanation
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:61: 185:85 note: expected type `rustc_const_math::ConstFloat`
.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.74/src/misc.rs:185:61: 185:85 note:    found type `f64`
error: aborting due to 3 previous errors
error: failed to compile `clippy v0.0.74`, intermediate artifacts can be found at `/tmp/cargo-install.VxmKaOYhFHxR`
```

PS. Sorry if I missed some guidelines on issues for the project, I didn't see anything in the guidelines.

**Comment by llogiq** [comment]
Can you retry with todays nightly?

**Comment by crumblingstatue** [comment]
> Can you retry with todays nightly?

Do you mean the one that *will* be released today?

The latest nightly currently available is `1.11.0-nightly (763f9234b 2016-06-06)`, with which this error happens.

**Comment by mcarton** [comment]
> PS. Sorry if I missed some guidelines on issues for the project, I didn't see anything in the guidelines.

It's normal, Clippy breaks once in a while because we use unstable *rustc* API. Thanks for the report, should be fixed in v0.0.75.
